Lib/test/test_typing.py

```diff
@@ -8756,6 +8756,16 @@ class B(str): ...
         self.assertIs(type(field_c2.__metadata__[0]), float)
         self.assertIs(type(field_c3.__metadata__[0]), bool)
 
+    def test_annotated_callable_returning_immutable(self):
+        class C:
+            def __setattr__(self, name, value):
+                raise Exception('should be ignored')
+
+        class A(str): ...
+
+        annotated = Annotated[C, A("A")]
+        self.assertIsInstance(annotated(), C)
+
```

Comment:
Could you possibly move this test a bit higher up in the file, so that it's next to the other tests that assert that instances of `_AnnotatedAlias` can be instantiated? I'm talking about these tests here:

https://github.com/python/cpython/blob/846fd721d518dda88a7d427ec3d2c03c45d9fa90/Lib/test/test_typing.py#L8471-L8503

You could also add a test for other kinds of generic aliases being instantiated. For example, this would trigger the same bug that you reported in the issue, even though it has nothing to do with `Annotated`:

```pycon
>>> class Foo[T]:
...     def __setattr__(self, attr, value):
...         raise RuntimeError("no")
...
>>> Foo[int]()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\alexw\AppData\Local\Programs\Python\Python312\Lib\typing.py", line 1142, in __call__
    result.__orig_class__ = self
    ^^^^^^^^^^^^^^^^^^^^^
  File "<stdin>", line 3, in __setattr__
RuntimeError: no
```

Reply:
addressed in 7289af40d46f34c92a3159de2a357b2a68c5269b and 7fcb3945ddb595eecad5c3fdd2be7af9a348d4a3